src/Report/ReportInterface.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Report;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Stopwatch\Stopwatch;
+
+interface ReportInterface
+{
+    /**
+     * Sets the Console Input.
+     *
+     * @param InputInterface
+     */
+    public function setInput(InputInterface $input);
+
+    /**
+     * Sets the Console Output.
+     *
+     * @param OutputInterface $output
+     */
+    public function setOutput(OutputInterface $output);
+
+    /**
+     * Sets the Stopwatch.
+     *
+     * @param Stopwatch $stopwatch
+     */
+    public function setStopwatch(Stopwatch $stopwatch);
+
+    /**
+     * Process changed files array.
+     *
+     * @param array $changed
+     */
+    public function process(array $changed);
```

Comment:
Pointless.
Keep it simple, keep it stupid. Have one thing for one responsibility. One for generating report (like report class), one for printing it (eg in current FixCommand)

It would simplify a lot your implementation and utests.